crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1889,37 +1883,40 @@ impl SymbolicationActor {
                     }
 
                     // Stackwalk the minidump.
-                    let cfi = load_cfi_for_processor(cfi_caches);
-                    let cfi = cfi
-                        .into_iter()
-                        .map(|(id, cache)| (id.into(), cache))
-                        .collect();
                     // we cannot map an `io::Error` into `MinidumpNotFound` since there is no public
                     // constructor on `ProcessResult`. Passing in an empty buffer should result in
                     // the same error though.
                     let minidump =
                         ByteView::open(minidump_path).unwrap_or_else(|_| ByteView::from_slice(b""));
-                    let process_state = ProcessState::from_minidump(&minidump, Some(&cfi))?;
+                    let process_state = {
+                        let cfi = load_cfi_for_processor(cfi_caches.clone())
+                            .into_iter()
+                            .map(|(id, cache)| (id.into(), cache))
+                            .collect();
+                        ProcessState::from_minidump(&minidump, Some(&cfi))?
+                    };
                     let minidump_state = MinidumpState::new(&process_state);
                     let object_type = minidump_state.object_type();
 
-                    let referenced_modules = process_state
+                    let missing_ids = process_state
                         .referenced_modules()
                         .into_iter()
                         .filter_map(|code_module| {
-                            Some((
-                                code_module.id().map(|cmid| cmid.into())?,
-                                object_info_from_minidump_module(object_type, code_module),
-                            ))
+                            code_module.id().and_then(|id| {
+                                let id = id.into();
+                                (!cfi_caches.contains_key(&id)).then(|| id)
```

Comment:
instead of cloning the whole `cfi_caches` (keys + values that we don’t need), maybe just collect the keys into a set. Then this just becomes the set difference. (referenced - loaded = missing)

Reply:
Yeah, that's a really good idea. I was losing sight of the forest for the trees working on this.